### PR TITLE
Reorganize C source into directories - v11

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/4609

Changes from last PR:
- Instead of rebasing, update my script.
- detect keywords now in detect/keywords

Redmine issue: https://redmine.openinfosecfoundation.org/issues/3166

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
- PR jasonish-pcap: https://buildbot.openinfosecfoundation.org/builders/jasonish-pcap/builds/457
- PR jasonish: https://buildbot.openinfosecfoundation.org/builders/jasonish/builds/814

### CODE REVIEW

File: src/alert-syslog.c [hunk 1]
```diff
@@ -35,9 +35,9 @@
 
 #include "detect.h"
 #include "detect-parse.h"
-#include "detect-engine.h"
-#include "detect-engine-mpm.h"
-#include "detect-reference.h"
+#include "detect/keywords/engine.h"
```
Comment (victorjulien) [inline src/alert-syslog.c]: `engine` is not a keyword. My idea was to distinguish between keywords and engine internals `src/detect/keywords/*` and `src/detect/engine/*`

**Comment by victorjulien** [comment]
Wonder if we should consider taking this a step further even, making it closer to the Rust app-layer "modules":
```
src/app-layer/modbus/mod.c
src/app-layer/modbus/parser.c
src/app-layer/modbus/detect.c
...
```
Af-packet:
```
src/source/af-packet/mod.c
src/source/af-packet/runmode.c
src/source/af-packet/capture.c
```
Thoughts? Other than "it's a lot of work ;)"

**Comment by jasonish** [comment]
> Thoughts? Other than "it's a lot of work ;)"

I think it makes sense, but its a lot of work, probably not as automated as this.  Whats here is 100% automated, which is why detect-engine got in there (it matched my test for keyword files which I'll need to tweak).

I think a layout like this might be better done manually..  Pick one, refactor, etc.

File: src/alert-syslog.c (hunk 1 above)
Comment (jasonish) [reply src/alert-syslog.c]: Fixed.

**Comment by jasonish** [comment]
Detect engine files fixed at https://github.com/OISF/suricata/pull/4757.

**Pull request closed without merging**